### PR TITLE
Gm focus

### PR DESCRIPTION
**Pull request by mgalbier**
- Add method to set NameChangeVisualizationFlags
- Execute scaling the first time immediately after adding a OnZoomHandler
- Prefer method before class for symbolMatcherPriorityList
- Change key binding for NameChangeFilter toggling
- Refresh focus information for every focus step
- Display an arrow connecting a node with its associated review comments
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77027601%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77027855%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77028014%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77028346%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23issuecomment-243831327%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77052929%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77052973%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77082278%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77084390%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23issuecomment-244004307%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23issuecomment-243831327%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Ok%2C%20that%27s%20all.%20Only%20very%20small%20things.%22%2C%20%22created_at%22%3A%20%222016-08-31T17%3A03%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks.%22%2C%20%22created_at%22%3A%20%222016-09-01T07%3A56%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2007a979352cd18241f3d2673bd29f9f93c8cf7448%20VisualizationBase/src/Scene.cpp%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77027601%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hmm%2C%20why%20is%20this%20change%20necessary%3F%20Wouldn%27t%20it%20perhaps%20make%20more%20sense%20to%20call%20the%20handling%20code%20from%20the%20outside%20in%20whatever%20case%20you%20need%20this%3F%20Logically%20it%20doesn%27t%20make%20sense%20to%20me%20to%20call%20this%20here%2C%20but%20maybe%20you%20could%20convince%20me%20otherwise.%22%2C%20%22created_at%22%3A%20%222016-08-31T16%3A59%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20added%20this%20to%20have%20the%20right%20scaling%20of%20components%20immediately%20after%20adding%20their%20OnZoomHandler.%20Without%20this%20change%20it%20would%20only%20take%20effect%20after%20the%20next%20zooming.%22%2C%20%22created_at%22%3A%20%222016-08-31T19%3A09%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%2C%20%7B%22body%22%3A%20%22There%20is%20one%20place%20where%20you%20call%20this%20method%2C%20and%20I%20think%20it%27s%20more%20correct%20for%20your%20to%20call%20the%20handler%20there%2C%20once%20you%27ve%20added%20it.%20The%20method%20here%20should%20not%20change.%22%2C%20%22created_at%22%3A%20%222016-08-31T22%3A04%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hmm%20yes%2C%20I%27ll%20change%20it.%22%2C%20%22created_at%22%3A%20%222016-08-31T22%3A20%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VisualizationBase/src/Scene.cpp%3AL606-613%22%7D%2C%20%22Pull%20af8c282dc7c3a985bb5cca7b3e929211b2d89edc%20CodeReview/src/commands/CFocus.cpp%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77028014%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20this%20needed%20so%20that%20the%20dev%20can%20make%20changes%20on%20the%20fly%3F%22%2C%20%22created_at%22%3A%20%222016-08-31T17%3A01%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20exactly.%22%2C%20%22created_at%22%3A%20%222016-08-31T19%3A09%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CodeReview/src/commands/CFocus.cpp%3AL158-168%22%7D%2C%20%22Pull%20817f582a45af81c55596803d0b47c79f8cc2c1c7%20CodeReview/src/CodeReviewManager.cpp%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77028346%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Sweet%2C%20this%20was%20indeed%20super%20easy.%22%2C%20%22created_at%22%3A%20%222016-08-31T17%3A03%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CodeReview/src/CodeReviewManager.cpp%3AL129-154%22%7D%2C%20%22Pull%2082065738053a956d065de9d8ef73d5847b83ea36%20VersionControlUI/src/commands/CDiff.cpp%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/440%23discussion_r77027855%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22make%20all%20of%20this%20nicer%20and%20one%20one%20line%3A%5Cr%5Cn%60QList%3CModel%3A%3ASymbolMatcher%3E%20symbolMatcherPriorityList%7B%5C%22Method%5C%22%2C%20%5C%22Class%5C%22%7D%3B%60%22%2C%20%22created_at%22%3A%20%222016-08-31T17%3A01%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/commands/CDiff.cpp%3AL119-127%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 07a979352cd18241f3d2673bd29f9f93c8cf7448 VisualizationBase/src/Scene.cpp 4'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/440#discussion_r77027601'>File: VisualizationBase/src/Scene.cpp:L606-613</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Hmm, why is this change necessary? Wouldn't it perhaps make more sense to call the handling code from the outside in whatever case you need this? Logically it doesn't make sense to me to call this here, but maybe you could convince me otherwise.
- <a href='https://github.com/mgalbier'></a> I added this to have the right scaling of components immediately after adding their OnZoomHandler. Without this change it would only take effect after the next zooming.
- <a href='https://github.com/dimitar-asenov'></a> There is one place where you call this method, and I think it's more correct for your to call the handler there, once you've added it. The method here should not change.
- <a href='https://github.com/mgalbier'></a> Hmm yes, I'll change it.
- [ ] <a href='#crh-comment-Pull 82065738053a956d065de9d8ef73d5847b83ea36 VersionControlUI/src/commands/CDiff.cpp 6'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/440#discussion_r77027855'>File: VersionControlUI/src/commands/CDiff.cpp:L119-127</a></b>
- <a href='https://github.com/dimitar-asenov'></a> make all of this nicer and one one line:
  `QList<Model::SymbolMatcher> symbolMatcherPriorityList{"Method", "Class"};`
- [ ] <a href='#crh-comment-Pull af8c282dc7c3a985bb5cca7b3e929211b2d89edc CodeReview/src/commands/CFocus.cpp 15'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/440#discussion_r77028014'>File: CodeReview/src/commands/CFocus.cpp:L158-168</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Is this needed so that the dev can make changes on the fly?
- <a href='https://github.com/mgalbier'></a> Yes, exactly.
- [ ] <a href='#crh-comment-Pull 817f582a45af81c55596803d0b47c79f8cc2c1c7 CodeReview/src/CodeReviewManager.cpp 29'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/440#discussion_r77028346'>File: CodeReview/src/CodeReviewManager.cpp:L129-154</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Sweet, this was indeed super easy.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/440#issuecomment-243831327'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Ok, that's all. Only very small things.
- <a href='https://github.com/dimitar-asenov'></a> Thanks.

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/440?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/440?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/440'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/440'>Refresh</a>.<a href='#crh-update-none'></a>

File: VisualizationBase/src/Scene.cpp [hunk 1]
```diff
@@ -606,6 +606,7 @@ void Scene::removeAllOnZoomHandlers()
 int Scene::addOnZoomHandler(OnZoomHandler onZoomHandler, OnZoomHandlerRemove onZoomHandlerRemove)
 {
 	onZoomHandlers_.insert(addOnZoomHandlerId_, {onZoomHandler, onZoomHandlerRemove});
+	onZoomHandler(mainViewScalingFactor_);
```
Comment (dimitar-asenov) [inline VisualizationBase/src/Scene.cpp]: Hmm, why is this change necessary? Wouldn't it perhaps make more sense to call the handling code from the outside in whatever case you need this? Logically it doesn't make sense to me to call this here, but maybe you could convince me otherwise.

File: VersionControlUI/src/commands/CDiff.cpp [hunk 2]
```diff
@@ -119,8 +119,8 @@ Interaction::CommandResult* CDiff::execute(Visualization::Item* source, Visualiz
 	versionB = unambigousPrefixPerRevision_.value(versionB, versionB);
 	QList<Model::SymbolMatcher> symbolMatcherPriorityList;
 
-	symbolMatcherPriorityList.append(Model::SymbolMatcher{"Class"});
 	symbolMatcherPriorityList.append(Model::SymbolMatcher{"Method"});
+	symbolMatcherPriorityList.append(Model::SymbolMatcher{"Class"});
```
Comment (dimitar-asenov) [inline VersionControlUI/src/commands/CDiff.cpp]: make all of this nicer and one one line:
`QList<Model::SymbolMatcher> symbolMatcherPriorityList{"Method", "Class"};`

File: CodeReview/src/commands/CFocus.cpp [hunk 3]
```diff
@@ -156,4 +158,10 @@ void CFocus::loadFocusInformation()
 	}
 }
 
+void CFocus::refreshFocusInformation()
+{
+	clearFocusInformation();
```
Comment (dimitar-asenov) [inline CodeReview/src/commands/CFocus.cpp:163]: Is this needed so that the dev can make changes on the fly?

File: CodeReview/src/CodeReviewManager.cpp [hunk 4]
```diff
@@ -128,17 +129,25 @@ NodeReviewsList* CodeReviewManager::loadReview(const VersionControlUI::DiffSetup
...
 
 	return nodeReviews_;
 }
 
+void CodeReviewManager::displayAndRegisterCodeReviewComment(Visualization::Item* associatedItem,
+																				NodeReviews* nodeReviews)
+{
+	auto overlay = new CodeReviewCommentOverlay{associatedItem, nodeReviews};
+	associatedItem->addOverlay(overlay, "CodeReviewComment");
+	auto arrowOverlay = new Visualization::ArrowOverlay{overlay, associatedItem,
```
Comment (dimitar-asenov) [inline CodeReview/src/CodeReviewManager.cpp:144]: Sweet, this was indeed super easy.

**Comment by dimitar-asenov** [comment]
Ok, that's all. Only very small things.

File: VisualizationBase/src/Scene.cpp (hunk 1 above)
Comment (mgalbier) [reply VisualizationBase/src/Scene.cpp]: I added this to have the right scaling of components immediately after adding their OnZoomHandler. Without this change it would only take effect after the next zooming.

File: CodeReview/src/commands/CFocus.cpp (hunk 3 above)
Comment (mgalbier) [reply CodeReview/src/commands/CFocus.cpp:163]: Yes, exactly.

File: VisualizationBase/src/Scene.cpp (hunk 1 above)
Comment (dimitar-asenov) [reply VisualizationBase/src/Scene.cpp]: There is one place where you call this method, and I think it's more correct for your to call the handler there, once you've added it. The method here should not change.

Comment (mgalbier) [reply VisualizationBase/src/Scene.cpp]: Hmm yes, I'll change it.

**Comment by dimitar-asenov** [comment]
Thanks.
